cmd/junitreport/pkg/parser/interfaces.go

```diff
@@ -0,0 +1,61 @@
+package parser
+
+import (
+	"bufio"
+
+	"github.com/openshift/origin/cmd/junitreport/pkg/api"
+)
+
+// TestOutputParser knows how to parse test output to create a collection of test suites
+type TestOutputParser interface {
+	Parse(input *bufio.Scanner) (*api.TestSuites, error)
+}
+
+// TestDataParser knows how to take raw test data and either characterize it as data that
+// contains useful information or extract the useful information from it
+type TestDataParser interface {
+	// ContainsIdentifier determines if the line contains the identifier for a test
+	ContainsIdentifier(line string) bool
+
+	// ExtractName extracts the name of the test case from test output lines and returns
+	// the name if all of the extracted names match. This extraction supports multiple lines
+	// as the test name often appears at the declaration and the test result line.
+	ExtractName(identifiers []string) (name string, ok bool)
```

Comment:
`ExtractTestName`? how is `ContainsIdentifier` different than `_, containsIdentifier := ExtractName()`?